File: src/main/java/me/Oxygen/modules/player/Teams.java
package me.Oxygen.modules.player;

import me.Oxygen.Oxygen;
import me.Oxygen.modules.Category;
import me.Oxygen.modules.Module;
import me.Oxygen.modules.ModuleRegister;
import net.minecraft.client.Minecraft;
import net.minecraft.entity.Entity;

@ModuleRegister(name = "Teams", category = Category.PLAYER)
public class Teams extends Module {
	
	public final static boolean isOnSameTeam(final Entity entity) {
		if(Oxygen.INSTANCE.ModMgr.getModule(Teams.class).isEnabled()) {
			if(Minecraft.getMinecraft().thePlayer.getDisplayName().getUnformattedText().startsWith("\247")) {
				if(Minecraft.getMinecraft().thePlayer.getDisplayName().getUnformattedText().length() <= 2
	                    || entity.getDisplayName().getUnformattedText().length() <= 2) {
					return false;
				}
				if(Minecraft.getMinecraft().thePlayer.getDisplayName().getUnformattedText().substring(0, 2).equals(entity.getDisplayName().getUnformattedText().substring(0, 2))) {
	                return true;
	            }
			}
		}
		return false;
	}
	
	/*public static boolean isInYourTeam(EntityLivingBase entity) {
        //mc.thePlayer ?: return false;

        if (scoreboardValue.get() &&(Minecraft.getMinecraft().thePlayer.team != null && entity.team != null &&
        		(Minecraft.getMinecraft().thePlayer.team.isSameTeam(entity.team));
            return true;

        if(armorColorValue.get()){
            val entityPlayer = entity as EntityPlayer
            if(Minecraft.getMinecraft().thePlayer.inventory.armorInventory[3] != null && entityPlayer.inventory.armorInventory[3] != null){
                val myHead = Minecraft.getMinecraft().thePlayer.inventory.armorInventory[3]
                val myItemArmor = myHead.item as ItemArmor


                val entityHead = entityPlayer.inventory.armorInventory[3]
                var entityItemArmor = myHead.item as ItemArmor

                if(myItemArmor.getColor(myHead) == entityItemArmor.getColor(entityHead)){
                    return true;
                }
            }
        }

        if (colorValue.get() && Minecraft.getMinecraft().thePlayer.displayName != null && entity.displayName != null) {
            val targetName = entity.displayName.formattedText.replace("§r", "");
            val clientName = Minecraft.getMinecraft().thePlayer.displayName.formattedText.replace("§r", "");
            return targetName.startsWith("§${clientName[1]}");
        }



        return false;
    }*/

}

File: src/main/java/cn/rainbow/oxygen/module/modules/player/Teams.kt
package cn.rainbow.oxygen.module.modules.player

import cn.rainbow.oxygen.Oxygen
import cn.rainbow.oxygen.module.setting.BooleanValue
import cn.rainbow.oxygen.module.Category
import cn.rainbow.oxygen.module.Module
import cn.rainbow.oxygen.module.ModuleInfo
import net.minecraft.client.Minecraft

import net.minecraft.entity.Entity
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.item.ItemArmor

@ModuleInfo(name = "Teams", category = Category.Player)
class Teams : Module() {

    companion object {
        private val armor = BooleanValue("Armor", false)

        @JvmStatic
        fun isOnSameTeam(entity: Entity): Boolean {
            val mc = Minecraft.getMinecraft()

            if (mc.thePlayer != null && !Oxygen.INSTANCE.moduleManager.getModule(Teams::class.java)!!.enabled)
                return false

            if (armor.currentValue) {
                if (entity is EntityPlayer && mc.thePlayer.inventory.armorInventory[3] != null &&
                    entity.inventory.armorInventory[3] != null) {
                    val myHead = mc.thePlayer.inventory.armorInventory[3]
                    val myItemArmor = myHead.item as ItemArmor
                    val entityHead = entity.inventory.armorInventory[3]
                    val entityItemArmor = myHead.item as ItemArmor
                    return myItemArmor.getColor(myHead) == entityItemArmor.getColor(entityHead)
                }
            }
            if (mc.thePlayer.displayName.unformattedText.startsWith("\u00a7")) {
                if (mc.thePlayer.displayName.unformattedText.length <= 2 ||
                    entity.displayName.unformattedText.length <= 2) {
                    return false
                }
                if (mc.thePlayer.displayName.unformattedText.substring(0, 2) ==
                    entity.displayName.unformattedText.substring(0, 2)) {
                    return true
                }
            }
            return false
        }
    }

}